chain/vendored/apply_transaction.go

```diff
@@ -67,6 +67,7 @@ func EVMApplyTransaction(msg *Message, config *params.ChainConfig, author *commo
 	receipt.GasUsed = result.UsedGas
 
 	// If the transaction created a contract, store the creation address in the receipt.
+	// TODO: This needs to be fixed
```

Comment:
This is the bug...the receipt's contract address is incorrectly set for deterministic deployments. 

We can pass in the `TestChainConfig` so that we get the overrides, but how do you identify whether this txn was for an override address? There is no contract Name or anything. You could re-hash the init bytecode (which is the calldata) and use that. Can't think of another solution.